File: 3371-harshad-number/solution.c
int sumOfTheDigitsOfHarshadNumber(int x) {
    int sum=0;
    int n=x;
    while(n>0){
        int r=n%10;
        sum+=r;
        n/=10;
    }
    if(x%sum==0)
    return sum;
    return -1;
}

File: 3371-harshad-number/solution.java
class Solution {
    public int sum(int n){
        int sum = 0;
        while( n != 0){
            sum += (n%10);
            n /= 10;
        }
        return sum;
    }
    public int sumOfTheDigitsOfHarshadNumber(int x) {
        int a = sum(x);
        if(x%a == 0) return a;
        return -1;
    }
}
